src/Config/ConfigResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Slim\Console\Config;
+
+use InvalidArgumentException;
+use ReflectionClass;
+use Slim\Console\Exception\ConfigNotFoundException;
+
+use function array_filter;
+use function file_exists;
+use function file_get_contents;
+use function implode;
+use function is_array;
+use function is_null;
+use function json_decode;
+use function strpos;
+
+class ConfigResolver
+{
+    public const FORMAT_JSON = 'json';
+    public const FORMAT_PHP = 'php';
+    //public const FORMAT_YAML = 'yaml';
+    private const CONFIG_FILENAME = 'slim-console.config';
+
+    /** @var string|null */
+    private $file = null;
+
+    /** @var string */
+    private $dir;
+
+    public function __construct(string $dir)
+    {
+        $this->dir = $dir;
+    }
+
+    public function loadConfig(): ?Config
+    {
+        $this->locateConfig();
+
+        return $this->loadConfigByFormat();
```

Comment:
Since ENV variables takes precedence over all other formats here what we should try first is `$this->attemptCreatingConfigFromEnv(): Config` which would call `Config::fromEnvironment()` (this method should throw something like `CannotCreateConfigFromEnvironmentException`. If we catch this, then we proceed to try and locate `PHP` config then `JSON` then `YAML`

Here we should do:

```php
try {
    return $this->attemptCreatingConfigFromEnv();
catch (CannotCreateConfigFromEnvironmentException $e) {
    return $this->attemptCreatingConfigFromSupportedFormats();
}
```